aws/resource_aws_sns_topic.go

```diff
@@ -415,6 +464,16 @@ func resourceAwsSnsTopicRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("sqs_failure_feedback_role_arn", aws.StringValue(attributeOutput.Attributes["SQSFailureFeedbackRoleArn"]))
 		d.Set("sqs_success_feedback_role_arn", aws.StringValue(attributeOutput.Attributes["SQSSuccessFeedbackRoleArn"]))
 
+		// set the boolean values
+		d.Set("fifo_topic", false)
```

Comment:
The docs say that once you set a topic type it can't be changed. Is there a way to ensure the topic type's not being changed during an update, during validation stage?

fifo_topic isn't an attribute available for [SetTopicAttributes](https://docs.aws.amazon.com/sns/latest/api/API_SetTopicAttributes.html)

Reply:
The better way to do that is to change the schema in the resource to have `ForceNew: true` on it.

So in here:
```
			"fifo_topic": {
				Type:     schema.TypeBool,
				Optional: true,
				Default:  false,
			},
```

Do this:

```
			"fifo_topic": {
				Type:     schema.TypeBool,
				Optional: true,
				Default:  false,
				ForceNew: true
			},
```

That way if that field is changed, terraform knows it needs to create a brand new topic to replace it, and you'll get a corresponding diff.

Reply:
Good call, added the schema change in https://github.com/hashicorp/terraform-provider-aws/pull/15828/commits/5019aadaf1c7dfefcbd89ce44fd0ca5380ed0881